prowler/providers/aws/aws_provider_new.py

```diff
@@ -0,0 +1,622 @@
+import os
+import pathlib
+import sys
+from dataclasses import dataclass
+from datetime import datetime
+from typing import Any, Optional
+
+from boto3 import client, session
+from botocore.config import Config
+from botocore.credentials import RefreshableCredentials
+from botocore.session import get_session
+from colorama import Fore, Style
+
+from prowler.config.config import aws_services_json_file
+from prowler.lib.check.check import list_modules, recover_checks_from_service
+from prowler.lib.logger import logger
+from prowler.lib.utils.utils import open_file, parse_json_file
+from prowler.providers.aws.config import (
+    AWS_STS_GLOBAL_ENDPOINT_REGION,
+    BOTO3_USER_AGENT_EXTRA,
+)
+from prowler.providers.aws.lib.arn.arn import parse_iam_credentials_arn
+from prowler.providers.aws.lib.credentials.credentials import (
+    create_sts_session,
+    validate_aws_credentials,
+)
+from prowler.providers.aws.lib.organizations.organizations import (
+    get_organizations_metadata,
+)
+from prowler.providers.common.provider import CloudProvider
+
+
+@dataclass
+class AWS_Organizations_Info:
+    account_details_email: str
+    account_details_name: str
+    account_details_arn: str
+    account_details_org: str
+    account_details_tags: str
+
+
+@dataclass
+class AWS_Credentials:
+    aws_access_key_id: str
+    aws_session_token: str
+    aws_secret_access_key: str
+    expiration: datetime
+
+
+@dataclass
+class AWS_Assume_Role:
```

Comment:
Can you please change that to `AWSAssumeRole`? To keep Python naming standards.

Reply:
Done